packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -1652,34 +1656,36 @@ export class CacheDataClient implements IDataClient {
           const compressionLevel: CompressionLevel =
             this.compressionDetails.compressionLevel;
           itemsToUse = await Promise.all(
-            itemsToUse.map(async ([key, value]) => {
-              return [key, await compressor.compress(compressionLevel, value)];
+            itemsToUse.map(async ([key, value, ttl]) => {
+              return [
+                key,
+                await compressor.compress(compressionLevel, value),
+                ttl,
+              ];
             })
           );
         }
       }
 
-      const ttlToUse = options?.ttl || this.defaultTtlSeconds;
       this.logger.trace(
         `Issuing 'setBatch' request; items length: ${
           itemsToUse.length
         }, ttl: ${ttlToUse.toString()}`
       );
-      return await this.sendSetBatch(cacheName, itemsToUse, ttlToUse);
+      return await this.sendSetBatch(cacheName, itemsToUse);
     });
   }
 
   private async sendSetBatch(
     cacheName: string,
-    items: [Uint8Array, Uint8Array][],
-    ttlSeconds: number
+    items: [Uint8Array, Uint8Array, number][]
   ): Promise<CacheSetBatch.Response> {
     const setRequests = [];
     for (const item of items) {
       const setRequest = new grpcCache._SetRequest({
         cache_key: item[0],
         cache_body: item[1],
-        ttl_milliseconds: ttlSeconds * 1000,
+        ttl_milliseconds: item[2] * 1000,
```

Comment:
For another day, we should extract the arithmetic in this file a single helper function.